### PR TITLE
Add `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by Nadrieril**
This adds a first attempt at contribution instructions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+- [How To Contribute](#how-to-contribute)
+- [Reporting Bugs](#reporting-bugs)
+- [Suggesting Enhancements](#suggesting-enhancements)
+- [Contributing Code](#contributing-code)
+
+
+## I Have a Question
+
+<!--
+> If you want to ask a question, we assume that you have read the available [Documentation]().
```
[the remark this answers is not in the document]
Comment (Nadrieril) [reply CONTRIBUTING.md:19]: This is commented out, we don't have documentation yet

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+#### Continuous Integration
+
+Charon runs checks in CI for each pull request. There are two kinds:
+- The `nix` check is the normal Charon CI that builds charon and runs the full test suite;
+- The `aeneas`, `eurydice` and `kyber` checks correspond to projects that use Charon that we endeavor to break as little as possible.
+
+Sometimes breakage is unavoidable however. In this case we can merge changes with these checks not passing. Before that:
+- For `aeneas`, we fix the issue ourselves. This involves making a PR to [Aeneas][] that fixes compilation on a best-effort basis (e.g. adding `raise` branches if there are new enum variants) and bumps their pin of charon (using `make update-charon-pin`). We request that the Aeneas change (excepting the pin update) be ready before we merge the change to Charon.
+- For `eurydice` and `kyber`, we discuss with their maintainers.
```
[the remark this answers is not in the document]
Comment (Nadrieril) [reply CONTRIBUTING.md:100]: The project maintainers. We don't want random contributors to go chat with Eurydice maintainers, do we?

[the remark this answers is not in the document]
Comment (Nadrieril) [reply CONTRIBUTING.md]: I added a precision

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+- Make sure that you are using the latest version of Charon;
+- Ensure you're using the same version of Charon to produce and consume llbc files;
+- Clean up cached files and recompile (inside the project, this can be done with `make clean; make`);
+- If relevant, recompile the project you use to consume llbc files (e.g. Aeneas of Eurydice);
+- Clean up leftover llbc files and regenerate them (inside the project, this can be done with `make clean-generated`);
+- Check if a similar issue has already been reported in the [issue tracker][].
+
+#### Reporting a Bug
+
+To be able to help you, we need the following information:
```
Comment (sonmarcho) [inline CONTRIBUTING.md:45]: We should put this in an issue template

**Review by sonmarcho — APPROVED** [review]
Looks good to me!
But I suggest waiting for the other reviews

**Comment by R1kM** [comment]
Looks good to me too, at least as an initial template (we can always extend later). Feel free to merge.

**Comment by sonmarcho** [comment]
I also added @msprotz as a reviewer

**Review by msprotz — APPROVED** [review]
👍

**Comment by sonmarcho** [comment]
The branch is good to merge.

File: CONTRIBUTING.md (hunk 3 above)
Comment (Nadrieril) [reply CONTRIBUTING.md:45]: Tracker in https://github.com/AeneasVerif/charon/issues/198

**Pull request merged**